src/structui/ui.py

```
@@ -282,17 +283,30 @@ def delete_current_container():
+                        val = getattr(e, 'value', getattr(getattr(e, 'sender', None), 'value', None))
                         if prop_type == 'integer' and val is not None and val != '':
                             try: val = int(float(val))
                             except ValueError: pass
                         elif prop_type in ['number', 'float'] and val is not None and val != '':
-                            try: val = float(val)
+                            try:
+                                if '.' in str(val):
+                                    val = float(val)
+                                else:
+                                    val = int(val)
                             except ValueError: pass
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)